AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClientOperations/AWSMobileClient+Credentials.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+// http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+import Foundation
+
+extension AWSMobileClient {
+    
+    /// Fetches the `AWSCredentials` asynchronously.
+    ///
+    /// - Parameter completionHandler: completionHandler which would have `AWSCredentials` if successfully retrieved, else error.
+    public func getAWSCredentials(_ completionHandler: @escaping(AWSCredentials?, Error?) -> Void) {
+        if self.internalCredentialsProvider == nil {
+            completionHandler(nil, AWSMobileClientError.cognitoIdentityPoolNotConfigured(message: "There is no valid cognito identity pool configured in `awsconfiguration.json`."))
```

Comment:
Should there be a return in this block?

Reply:
These changes are just copied from our old implementation. But looks like you are right, will add a return